src/components/BackButton.tsx

```diff
@@ -1,41 +1,54 @@
 import { useTranslation } from "next-i18next";
 import React, { CSSProperties } from "react";
-import  { ArrowBackOutlined }  from "@mui/icons-material";
+import { ArrowBackOutlined } from "@mui/icons-material";
 import { useRouter } from "next/router";
 import colors from "../styles/colors";
 
-function BackButton({ style, callback }: { style?: CSSProperties; callback?: () => void }) {
+interface BackButtonProps {
+    style?: CSSProperties;
+    callback?: () => void;
+    isVisible?: boolean;
+}
+
+const BackButton: React.FC<BackButtonProps> = ({
+    style,
+    callback,
+    isVisible = true,
+}) => {
     const { t } = useTranslation();
     const router = useRouter();
-    const pathname = router.pathname || "";
+    const pathname = router?.pathname || "";
 
-    if (pathname !== "/" && pathname !== "/home-page") {
-        return (
-            <a
-                style={{
-                    display:"flex",
-                    alignContent:"center",
-                    fontWeight: "bold",
-                    color: colors.secondary,
-                    ...style
-                }}
-                data-cy="testBackButton"
-                onClick={() => {
-                    if (callback) {
-                        callback();
-                    } else {
-                        // TODO: check if the previous page in history is from Aletheia
```

Comment:
nitpick (non-blocking): if you remove a comment, make sure we at least have an issue to track the work here. Ideally, this should still remain in the codebase.

Reply:
I was trying to introduce an interface to add a parameter needed for testing a component, but I didn’t run the tests outside the test interface. As a result, I didn’t notice the change in logic in this part. I will redo this section and keep the back button component as it was, performing the tests and adding the required parameter in a different way. Please hold off on merging for now.